src/main/scala/is/hail/methods/MinHash.scala

```diff
@@ -0,0 +1,64 @@
+package is.hail.methods
+
+import breeze.linalg.{DenseMatrix, DenseVector, min}
+import is.hail.variant.{Genotype, Variant, VariantDataset}
+import is.hail.annotations.Annotation
+import scala.collection.mutable.HashSet
+import scala.util.hashing.{MurmurHash3 => MH3}
+import scala.util.Random
+
+object MinHash {
+  // returns a k by vds.nSamples matrix
+  def kMinHash(vds: VariantDataset, k: Int): DenseMatrix[Int] = {
+    val n = vds.nSamples
+    def seqOp(l: DenseMatrix[Int], r: (Variant, (Annotation, Iterable[Genotype]))): DenseMatrix[Int] = r match {
+      case (_, (_, gs)) =>
+        val gsItr = gs.hardCallGenotypeIterator
+        val hashes = DenseVector.fill[Int](k)(Random.nextInt())
+        var i = 0
+        while(i < n) {
+          //TODO: Write a min UFunc that modifies the left arg in place
```

Comment:
Rather than UFunc, once we get to stage of optimizing performance, I suspect we'll just want to aggregate into an Array[Int] (representing the "vectorized matrix") rather than DenseMatrix[Int], and turn it into a DenseArray[Int] only at the very end.  You can still write abstractions to make this clean, but I suspect working through DenseMatrix introduces a good amount of overhead (e.g., accessing and writing elements involves a lot of checks).